tests/nvme_copy_test.py

```diff
@@ -4,56 +4,124 @@
 #
 # Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved.
 #
-# Author: Arunpandian J <apj.arun@samsung.com>
+# Authors: Arunpandian J <apj.arun@samsung.com>
+#          Joy Gu <jgu@purestorage.com>
 
 """
 NVMe Copy Testcase:-
 
     1. Issue copy command on set of block; shall pass.
+    2. If cross-namespace copy formats are supported, enable and test
+       cross-namespace copy formats.
 
 """
 
+import subprocess
+
 from nvme_test import TestNVMe
 
 
 class TestNVMeCopy(TestNVMe):
 
     """
-    Represents NVMe Verify testcase.
+    Represents NVMe Copy testcase.
         - Attributes:
-              - start_block :   starting block of to verify operation.
-              - range :         Range of blocks for DSM operation.
-              - slbs :          64-bit addr of first block per range
+              - ocfs : optional copy formats supported
+              - host_behavior_data : host behavior support data to restore during teardown
               - test_log_dir :  directory for logs, temp files.
     """
 
     def setUp(self):
         """ Pre Section for TestNVMeCopy """
         super().setUp()
-        self.start_block = 0
-        self.range = 1
-        self.slbs = 1
-        self.namespace = 1
+        print("\nSetting up test...")
+        self.ocfs = self.get_ocfs()
+        cross_namespace_copy = self.ocfs & 0xc
+        if cross_namespace_copy:
+            # get host behavior support data
+            get_features_cmd = "nvme get-feature " + self.ctrl + \
+                    " --feature-id=0x16 -l 512 -b"
+            print("Running command:", get_features_cmd)
+            proc = subprocess.Popen(get_features_cmd,
+                                    shell=True,
+                                    stdout=subprocess.PIPE)
+            self.host_behavior_data = proc.stdout.read()
+            # enable cross-namespace copy formats
+            data = list(self.host_behavior_data)
```

Comment:
Why is the `list` conversion necessary? Should be possible to index a `bytes` object directly?

Reply:
```
TypeError: 'bytes' object does not support item assignment
```

Reply:
Yes, so that was my suggestion below; you can reassemble the new bytes with the updated value:
```py
data = data[:4] + cross_namespace_copy.to_bytes(2, 'little') + data[6:]
```
It would be preferable to avoid iterating byte-by-byte twice.